pkg/cmd/pulumi/crypto_local.go

```diff
@@ -35,8 +37,20 @@ func readPassphrase(prompt string) (string, error) {
 	if phrase, ok := os.LookupEnv("PULUMI_CONFIG_PASSPHRASE"); ok {
 		return phrase, nil
 	}
+	if phraseFile, ok := os.LookupEnv("PULUMI_CONFIG_PASSPHRASE_FILE"); ok {
+		phraseFilePath, err := filepath.Abs(phraseFile)
+		if err != nil {
+			return "", errors.New("unable to construct a path the PULUMI_CONFIG_PASSPHRASE_FILE")
+		}
+		phraseDetails, err := ioutil.ReadFile(phraseFilePath)
```

Comment:
What if the file has trailing newlines? I assume we do not want to treat that as part of the passphrase.

Maybe do it like how we read from the console elsewhere (which can be from a file via redirected std input)?

https://github.com/pulumi/pulumi/blob/848a98d5112ec93239501fd66b51c9f1f30353ac/sdk/go/common/util/cmdutil/console.go#L72-L78